### PR TITLE
addressed #40 issue and changed the way choosing recipe with sampleRate

### PR DESCRIPTION
**Pull request by allmywant**

### CODE REVIEW

**Comment by allmywant** [comment]
For #40 issue, I kept existing inTest and getRecipe functions and added a ```getStoredDecision``` function.

**Comment by allmywant** [comment]
Just found something forgot to change, will submit another commit in this PR.

**Comment by allmywant** [comment]
@kingo55 @dapperdrop 
It's now ready for reviewing.

File: lib/mojito.js [hunk 1]
```diff
@@ -410,117 +382,63 @@ Mojito = (function ()
...
-                        return false;
-                    }
 
-                    this.setCookie('', values[0] + '.' + recipe);
+                var storedDecision = this.getStoredDecision();
 
-                    return true;
+                if (storedDecision.recipe !== recipe)
+                {
+                    this.setCookie('', (storedDecision?'1':'0') + '.' + recipe);
```
Comment (kingo55) [inline lib/mojito.js]: Nitpicking here - Is there ever a case where you would call `setRecipe()` but `inTest === 0`?

**Review by dapperdrop — APPROVED** [review]

**Comment by kingo55** [comment]
AA test (40-30-30 split) with this code is looking solid @allmywant - I might keep it up and running a little longer. No one ever complains about more data :P...

image

Screen Shot 2019-12-24 at 8 52 22 pm

File: lib/mojito.js (hunk 1 above)
Comment (allmywant) [reply lib/mojito.js]: @kingo55 
You're right. There is no cases, so, it's safe to change it to:
```this.setCookie('',  '1.' + recipe);```
I will change it.

Comment (allmywant) [reply lib/mojito.js]: Changed.

Comment (kingo55) [reply lib/mojito.js]: Awesome... thanks @allmywant !

**Review by kingo55 — APPROVED** [review]

**Pull request merged**